### ISSUE TITLE
tabular data/ noisy instances/ new datasets

### ISSUE THREAD

**Issue by nazaretl** [zhangchbin/OnlineLabelSmoothing#7]
Hi,
thanks for sharing your implementation. I have some questions about it:

1.	Does it also work on tabular data?
2.	Is the code tailored to the datasets used in the paper or can one apply it to any data?
3.	Is it possible to identify the noisy instances (return the noisy IDs or the clean set)?

Thanks!

**Comment by zhangchbin** [comment]
> Hi, thanks for sharing your implementation. I have some questions about it:
> 
> 1. Does it also work on tabular data?
> 2. Is the code tailored to the datasets used in the paper or can one apply it to any data?
> 3. Is it possible to identify the noisy instances (return the noisy IDs or the clean set)?
> 
> Thanks!

Hi, @nazaretl 
1. I'm sorry that we don't explore our method on tabular data. I think it will work if there are similarities between different classes. 
2. I think the code can be applied in any classification dataset. But there are three hyper-parameters to consider: 1) the weighted factor between the hard loss and the soft loss. 2) the phase of updating soft labels. 3) 
maximum confidence for the correct class in the soft labels. In our experiments, we simply set the 1) weighted factor to 0.5 and set 2) the phase to one epoch. But there are some differences between CIFAR and Fine-grained datasets. Specifically, we set the maximum confidence as 1.0 on CIFAR or ImageNet. But we find that the optimization is too fast on 4 fine-grained datasets due to the ImageNet pre-trained. So we set the maximum confidence to 0.9 on 4 fine-grained datasets. 
You'd better fine-tune these hyper-parameters on other datasets.
3. Simply, after training, you can use the model to infer the class of training images. If the model predict the class with high confidence and the class is different from the GT, the sample is noisy.

**Comment by nazaretl** [comment]
many thanks for your response!
